h5py/tests/hl/test_vds/test_highlevel_vds.py

```diff
@@ -267,5 +267,59 @@ def test_slice_source(self):
     def tearDown(self):
         shutil.rmtree(self.tmpdir)
 
+@ut.skipUnless(vds_support,
+               'VDS requires HDF5 >= 1.9.233')
+class IndexingTestCase(ut.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp()
+        # Create source file (1.h5)
+        with h5.File(osp.join(self.tmpdir, '1.h5'), 'w') as f:
+            d = f.create_dataset('data', (10,), 'i4')
+            d[:] = np.arange(10)*10
+
+    def test_index_layout(self):
+        # Assemble virtual dataset (indexing target)
+        layout = h5.VirtualLayout((100,), 'i4')
+
+        inds = [3,6,20,25,33,47,70,75,96,98]
+
+        filename = osp.join(self.tmpdir, "1.h5")
+        vsource = h5.VirtualSource(filename, 'data', shape=(10,))
+        layout[inds] = vsource
+
+        outfile = osp.join(self.tmpdir, 'VDS.h5')
+
+        # Assembly virtual dataset (indexing source)
+        layout2 = h5.VirtualLayout((5,), 'i4')
+
+        inds2 = [0,1,4,5,8]
+        layout2 = vsource[inds2]
+
+        # Add virtual datasets to output file and close
+        with h5.File(outfile, 'w', libver='latest') as f:
+            f.create_virtual_dataset('/data', layout, fillvalue=-5)
+            f.create_virutal_dataset('/data2', layout2, fillvalue=-3)
+
+        # Read data from virtual datasets
+        with h5.File(outfile, 'r') as f:
+            data = f['/data'][()]
+            data2 = f['/data2'][()]
+
+        # Verify
+        assert_array_equal(data[inds], np.arange(10)*10)
+        assert_array_equal(data2, [0,10,40,50,80])
+
+        mask = np.zeros(100)
+        mask[inds] = 1
+        self.assertEqual(data[mask == 0].min(), -5)
+        self.assertEqual(data[mask == 0].max(), -5)
+
+        self.assertEqual(data2[(2,3,6,7,9)].min(), -3)
+        self.assertEqual(data2[(2,3,6,7,9)].max(), -3)
```

Comment:
I don't think these checks make sense. You've filled the whole virtual dataset for `data2` with selected values from the source, so the fill value shouldn't appear at all.

Reply:
Yes apologies, I don't actually have a h5py development environment set up, so I am trying to write tests without actually running them.